### PR TITLE
Combination feature desactivation

### PR DESCRIPTION
**Pull request by aleeks**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Fix "attributes & features" tabs display & toolbar<br /> Fix "combination" bloc on "new product" page |
| Type? | bug fix |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1227 |
| How to test? | \- With a correct install (no corrupted product/combination), change combination & feature settings (BO > performance),<br /> - Go to "attributes & features tabs", test if is ok with settings, <br /> - Go on "new product" page and see "declinaison bloc" on the right is active or not |

### CODE REVIEW

File: src/Adapter/Configuration.php [hunk 1]
```diff
@@ -75,4 +75,22 @@ public function set($key, $value)
 
         return $this;
     }
+
+    /**
+     * Return if Feature feature is active or not
+     * @return bool
+     */
+    public function featureIsActive()
```
Comment (mickaelandrieu) [inline src/Adapter/Configuration.php]: why theses functions ? you don't use it.

Comment (mickaelandrieu) [inline src/Adapter/Configuration.php]: Except if I'm wrong, it's not used, right ?

File: controllers/admin/AdminAttributesGroupsController.php [hunk 2]
```diff
@@ -505,17 +504,19 @@ public function initContent()
...
-                'href' => self::$currentIndex.'&addattribute_group&token='.$this->token,
-                'desc' => $this->trans('Add new attribute', array(), 'Admin.Catalog.Feature'),
-                'icon' => 'process-icon-new'
-            );
-            $this->page_header_toolbar_btn['new_value'] = array(
-                'href' => self::$currentIndex.'&updateattribute&id_attribute_group='.(int)Tools::getValue('id_attribute_group').'&token='.$this->token,
-                'desc' => $this->trans('Add new value', array(), 'Admin.Catalog.Feature'),
-                'icon' => 'process-icon-new'
-            );
+        if (Combination::isFeatureActive()) {
```
Comment (mickaelandrieu) [inline controllers/admin/AdminAttributesGroupsController.php]: could be probably be declared as a private attribute.

**Comment by vincentbz** [comment]
Tested and ok, thanks !

No combination possible when option is disabled
image

The attribute page is disabled too
image

File: controllers/admin/AdminFeaturesController.php [hunk 3]
```diff
@@ -190,18 +190,20 @@ public function renderForm()
 
     public function initPageHeaderToolbar()
     {
-        if (empty($this->display)) {
-            $this->page_header_toolbar_btn['new_feature'] = array(
-                'href' => self::$currentIndex.'&addfeature&token='.$this->token,
-                'desc' => $this->trans('Add new feature', array(), 'Admin.Catalog.Feature'),
-                'icon' => 'process-icon-new'
-            );
+        if (Feature::isFeatureActive()) {
```
Comment (mickaelandrieu) [inline controllers/admin/AdminFeaturesController.php]: always the same pattern, how about a new function containing:

```php
return Feature::isFeatureActive() && empty($this->display);
```

File: src/Adapter/Configuration.php (hunk 1 above)
Comment (aleeks) [inline src/Adapter/Configuration.php]: No, it's not used for the moment. As I was working on combinaison and feature, I make 2 functions

File: controllers/admin/AdminAttributesGroupsController.php (hunk 2 above)
Comment (mickaelandrieu) [inline controllers/admin/AdminAttributesGroupsController.php]: always the same pattern, how about a new function containing:

```php
return Feature::isFeatureActive() && empty($this->display);
```

File: controllers/admin/AdminFeaturesController.php (hunk 3 above)
Comment (aleeks) [inline controllers/admin/AdminFeaturesController.php]: It's not really the same : 
- sometimes $this->display == 'something', sometimes empty($this->display) 
- Combination:: vs Feature::

**Comment by mickaelandrieu** [comment]
Thank you @aleeks

**Pull request merged**